### PR TITLE
PERF: no need to check for DataFrame in pandas.core.computation.expressions

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
In practice, this case is only covered by specific tests using DataFrame in `test_expressions.py`, not from actual usage through ops (so I just updated the failing tests to not use any DataFrame as input to those checks). 
Although our test suite might not be the best suited to check that, since many tests are small and thus don't use the numexpr path... (fixing that in https://github.com/pandas-dev/pandas/pull/40463)

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
cc @jreback you might be more familiar with the history behind this. I suppose at some point, DataFrames were passed directly to `evaluate` in `expressions.py`. But now we are only passing arrays from `array_ops.py`, so I just updated the tests.

File: pandas/tests/test_expressions.py [hunk 1]
```diff
@@ -170,16 +173,9 @@ def testit():
 
             op = getattr(operator, opname)
 
-            result = expr._can_use_numexpr(op, op_str, left, left, "evaluate")
```
Comment (jreback) [inline pandas/tests/test_expressions.py:173]: so this will just raise now? if so can you add a test

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_expressions.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/tests/test_expressions.py:173]: Actually, it seems we just return False. So I could leave that as a test, but note that this is a file-internal helper function (leading underscore) that's within pandas never called with a DataFrame. So I would basically be testing something that is not expected to have any specific behaviour.

**Comment by jorisvandenbossche** [comment]
@jreback OK here?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jorisvandenbossche

**Pull request merged**